### PR TITLE
Run Without System Tray Support

### PR DESCRIPTION
**Pull request by TafThorne**
When the system tray feature is not present, quietly continue to operate.  As this is no longer considered a critical feature we spend less time waiting for it to become available.  We know it will not be available on a number of GNU Linux desktop setups going forward.

This is a response to #155 and maybe sufficient to conclude the required work there.

### CODE REVIEW

**Comment by TafThorne** [comment]
To me this looks like CI system issue and not something that I have broken.  
AppVeyor Failure

I am not familiar with AppVeyor though so I this is an operativity to look into what it is and is trying to do.  I shall keep my fingers crossed for travis-ci behaving as it did on my fork.

**Comment by p12tic** [comment]
Going forward, I wonder if Barrier could check for tray support asynchronously and be usable right away.

With the current iteration, does barrier silently work if the window is closed?

**Comment by AdrianKoshka** [comment]
Does GNOME have it's own tray?

**Comment by TafThorne** [comment]
> With the current iteration, does barrier silently work if the window is closed?

@p12tic see #155 where it seems to have been asserted that it will continue to operate after closure.

**Comment by TafThorne** [comment]
> Does GNOME have it's own tray?

TLDR: No

It is GNOME that is getting rid of the tray AFAIK.  See https://blogs.gnome.org/aday/2017/08/31/status-icons-and-gnome/ 

They suggest an addon that you can install if you really, really want the notifcation area/status icons/system tray back.  Sadly that addon is not Ubuntu 18/04 compatible at the moment and the project is reported to be on hold.

**Comment by AdrianKoshka** [comment]
So, what do we do?

**Comment by TafThorne** [comment]
> So, what do we do?

Operate without a system tray when there is no support for one present as discussed under #155
